sda-commons-server-s3-testing/src/main/java/org/sdase/commons/server/s3/testing/S3Extension.java

```diff
@@ -0,0 +1,117 @@
+package org.sdase.commons.server.s3.testing;
+
+import java.io.File;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.jupiter.api.extension.AfterAllCallback;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.sdase.commons.server.s3.testing.builder.ContentObject;
+import org.sdase.commons.server.s3.testing.builder.FileObject;
+import org.sdase.commons.server.s3.testing.builder.MockObject;
+import org.sdase.commons.server.s3.testing.builder.StreamObject;
+
+/**
+ * JUnit 5 extension for running a AWS S3-compatible object storage alongside the (integration)
+ * tests. Use {@link #getEndpoint()} to retrieve the endpoint URL to connect to.
+ *
+ * <p>Example usage:
+ *
+ * <pre>
+ *   &#64;RegisterExtension
+ *   static final S3Extension S3_EXTENSION = S3Extension.builder().build();
+ * </pre>
+ */
+public class S3Extension extends S3InMemoryMock implements BeforeAllCallback, AfterAllCallback {
```

Comment:
Open for discussion as in the PR for OPA/Auth:
Do we want a common naming convention according to Wiremock? We could rename the extension to "S3ClassExtension" because it implements the "BeforeAll" and "AfterAll" callbacks.